server/api/src/main/scala/com/prisma/api/database/mutactions/mutactions/DeleteDataItemNested.scala

```diff
@@ -0,0 +1,26 @@
+package com.prisma.api.database.mutactions.mutactions
+
+import com.prisma.api.database._
+import com.prisma.api.database.mutactions.{ClientSqlDataChangeMutaction, ClientSqlStatementResult, MutactionVerificationSuccess}
+import com.prisma.api.mutations.NodeSelector
+import com.prisma.api.schema.APIErrors
+import com.prisma.shared.models.Project
+import slick.jdbc.MySQLProfile.api._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{Failure, Success, Try}
+
+case class DeleteDataItemNested(project: Project, where: NodeSelector) extends ClientSqlDataChangeMutaction {
```

Comment:
Could we directly substitute the uses of `DeleteDataItem` with this one? This one looks better to me.

Reply:
We can probably do that at a later point when we do not discern between top-level and nested mutactions that heavily anymore. At the moment that change would require a lot of extra refactoring.